src/waveform/renderers/waveformmarkset.cpp

```diff
@@ -35,9 +35,9 @@ void WaveformMarkSet::setup(const QString& group, const QDomNode& node,
             WaveformMarkPointer pMark(new WaveformMark(group, child, context, signalColors));
 
             bool uniqueMark = true;
-            if (pMark->m_pPointCos) {
+            if (!pMark->is_m_pPointCos_Null()) {
```

Comment:
It seems here that it will not be possible(or easy) to put this code in getConfigKey(), as the item object of QString is created in the block of if, so who to do it, I can be difficult, moreover it seems that if we have to do this, then there is the change of the flags like uniqmask, need the variable controlItemSet, and qwarning function of that class will also be used, which can be difficult, generalization of the function seems to be lost in that case, but if there is some way please tell, It seems to be is_m_pPointCos() is not too bad, except it tells some info about its private object, and if in the constructor of the WaveformMark class, we set the default value of m_pPointCos to null then it will make some sence